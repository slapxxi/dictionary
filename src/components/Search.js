// @flow
import React, { Component } from 'react';
import glamorous from 'glamorous';
import { theme } from '../lib/constants';

type Props = { value: string, onChange?: (string) => void };

class Search extends Component<Props> {
  handleChange = ({
    currentTarget,
  }: SyntheticEvent<HTMLInputElement>) => {
    const { onChange } = this.props;
    if (onChange) {
      onChange(currentTarget.value);
    }
  };

  render() {
    const { value } = this.props;
    return (
      <Input
        className="search"
        value={value}
        data-test="input"
        onChange={this.handleChange}
        placeholder="Search"
      />
    );
  }
}

const Input = glamorous.input({
  display: 'block',
  width: '100%',
  border: 0,
  backgroundColor: 'transparent',
  fontSize: 64,
  textAlign: 'center',
  color: theme.text,
  '&:focus': {
    outline: 0,
    '&::placeholder': {
      color: 'transparent',
    },
  },
  '::placeholder': {
    color: theme.subtext,
  },
});

export default Search;
