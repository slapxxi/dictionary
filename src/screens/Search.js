import React, { Component } from 'react';
import glamorous from 'glamorous';
import { Button } from '../components';
import { theme } from '../lib/constants';

class Search extends Component {
  state = { query: '', category: 'projects' };

  handleChange = ({ target }) => {
    this.setState({ query: target.value });
  };

  handleSearch = () => {};

  render() {
    return (
      <Container>
        <Heading>Search</Heading>
        <InputContainer>
          <Input
            className="search"
            type="search"
            placeholder="Query..."
            value={this.state.query}
            onChange={this.handleChange}
          />
        </InputContainer>
        <Button onPress={this.handleSearch}>Search</Button>
      </Container>
    );
  }
}

const Container = glamorous.div({
  display: 'flex',
  flexDirection: 'column',
  flex: 1,
  padding: 20,
});

const Heading = glamorous.h1({
  fontWeight: 'normal',
  color: theme.heading,
});

const InputContainer = glamorous.div({
  flex: 1,
  paddingTop: 20,
  paddingBottom: 20,
});

const Input = glamorous.input({
  appearance: 'none',
  display: 'inline-block',
  width: '100%',
  padding: 10,
  fontSize: 18,
  border: '1px solid lightgrey',
  borderRadius: 4,
});

export default Search;
