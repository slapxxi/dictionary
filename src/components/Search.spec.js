import React from 'react';
import { shallow, render } from 'enzyme';
import Search from './Search';

it('renders', () => {
  const component = render(<Search value="test" />);
  expect(component).toMatchSnapshot();
});

it('triggers onChange when modified', () => {
  const change = jest.fn();
  const component = shallow(<Search onChange={change} />);
  component
    .find('[data-test="input"]')
    .simulate('change', { currentTarget: { value: 'test' } });
  expect(change).toBeCalledWith('test');
});
