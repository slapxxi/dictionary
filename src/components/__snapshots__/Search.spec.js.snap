// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.glamor-0,
[data-glamor-0] {
  display: block;
  width: 100%;
  border: 0;
  background-color: transparent;
  font-size: 64px;
  text-align: center;
  color: white;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  outline: 0;
}

.glamor-0:focus::placeholder,
[data-glamor-0]:focus::placeholder {
  color: transparent;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  color: #6D6F7f;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  color: #6D6F7f;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  color: #6D6F7f;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  color: #6D6F7f;
}

<input
  class="glamor-0 search"
  data-test="input"
  placeholder="Search"
  value="test"
>
  
</input>
`;
